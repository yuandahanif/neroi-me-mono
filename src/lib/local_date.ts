const local_date = (
  date: string | Date,
  options?: {
    timeStyle?: "full" | "long" | "medium" | "short";
    dateStyle?: "full" | "long" | "medium" | "short";
  }
) => {
  let d;

  if (typeof date === "string") {
    d = new Date(date);
  } else {
    d = date;
  }

  const formater = new Intl.DateTimeFormat("id", {
    dateStyle: options?.dateStyle ?? "full",
    timeStyle: options?.timeStyle ?? "full",
  });

  return formater.format(d);
};

export default local_date;
