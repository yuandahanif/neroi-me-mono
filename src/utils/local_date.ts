const local_date = (date: string) => {
  const d = new Date(date);
  const formater = new Intl.DateTimeFormat("id", {
    dateStyle: "full",
    // timeStyle: "short",
  });
  return formater.format(d);
};

export default local_date;
