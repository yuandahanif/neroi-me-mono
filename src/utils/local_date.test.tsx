import { render } from "@testing-library/react";
import local_date from "./local_date";

describe("local_date function", () => {
  it("should return the same value from the same input", () => {
    const { container } = render(
      <span>{local_date("2023-04-13T17:57:36.397Z")}</span>
    );

    expect(local_date("2023-04-13T17:57:36.397Z")).toBe("Jumat, 14 April 2023");

    expect(local_date("2023-04-14T17:57:36.397Z")).toBe("Sabtu, 15 April 2023");

    expect(local_date("2023-04-13T17:57:36.397Z")).toBe("Jumat, 14 April 2023");

    // screen.debug();
    // expect(container).toMatchSnapshot();
  });

  it("should handle string parameter", () => {
    expect(local_date("2023-04-13T17:57:36.397Z")).toBe("Jumat, 14 April 2023");
  });

  it("should handle date parameter", () => {
    expect(local_date(new Date(2023, 6, 21))).toBe("Jumat, 21 Juli 2023");
  });
});
