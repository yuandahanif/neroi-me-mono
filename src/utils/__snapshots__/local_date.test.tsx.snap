// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`local_date function should return the same value from the same input 1`] = `
<div>
  <span>
    Jumat, 14 April 2023 00.57
  </span>
</div>
`;
